src/orion/core/io/cmdline_parser.py

```diff
@@ -0,0 +1,229 @@
+=============================================================================
+
+.. module:: cmdline_parser
+   :platform: Unix
+   :synopsis: Parsing and building of the command line input for using script
+
+Simplify the parsing of a command line by storing every values inside an `OrderedDict`
+mapping the name of the argument to its value as a key-pair relation. Positional arguments
+are stored in the format `_X` where `X` represent the index of that argument inside
+the command line.
+
+CmdlineParser provides an interface to parse command line arguments from input but also
+facilities to build it again as a list or an already formatted string.
+"""
+from collections import OrderedDict
+import os
+
+
+class CmdlineParser(object):
+    """Simple class for commandline arguments parsing.
+
+    CmdlineParser aims at providing a simple class to interpret commandline arguments
+    for the purposes of Orion. It can transform a list of string representing arguments to their
+    corresponding values. It can also recreate that string from the values by maintaing a template
+    of the way the arguments were passed.
+
+    Parameters
+    ----------
+    parsing_tokens : dict, optional
+        Dictionary containing the tokens to use when parsing `-` and `_`.
+
+    Attributes
+    ----------
+    arguments : OrderedDict
+        Commandline arguments' name and value(s).
+    template : list
+        List of template-ready strings (in the form 'something_{0}') for formatting.
+
+    """
+
+    def __init__(self, parsing_tokens=None):
+        """See `CmdlineParser` description"""
+        self.arguments = OrderedDict()
+        self._already_parsed = False
+        self.template = []
+
+        parsing_tokens = {'-': '??', '_': '!!'}
+
+        if parsing_tokens is not None:
+            parsing_tokens.update(parsing_tokens)
+
+        self._dash_token = parsing_tokens['-']
+        self._underscore_token = parsing_tokens['_']
+
+    def format(self, configuration):
+        """Format the current template.
+
+        Recreate the string of argument using the template made at parse-time and
+        the values inside the `configuration` argument.
+
+        Parameters
+        ----------
+        configuration : dict
+            Dictionary storing the keys and values to be passed to the `format` function.
+            This would typically be `parser.arguments` where `parser` is a `CmdlineParser`
+            instance.
+
+        Returns
+        -------
+        str
+            A recreated string of the commandline passed to Orion.
+
+        """
+        return " ".join(self.template).format(**configuration)
+
+    def parse(self, commandline):
+        """Parse the `commandline` argument.
+
+        Create an OrderedDict where the keys are the names of the arguments and the values
+        are the actual values of the each argument. The arguments can be a single value or a
+        list of values. This also supports positional arguments.
```

Comment:
It also creates a template for the `format()` method.

Reply:
Done.